ciphers/shuffled_shift_cipher.py

```diff
@@ -0,0 +1,173 @@
+        key_list_options = (
+            string.ascii_letters + string.digits + string.punctuation + " \t\n"
+        )
+
+        keys_l = []
+
+        # creates points known as breakpoints to break the key_list_options at those points and pivot each substring
+        breakpoints = sorted(set(self.__passcode))
+        temp_list = []
+
+        # algorithm for creating a new shuffled list, keys_l, out of key_list_options
+        for i in key_list_options:
+            temp_list.extend(i)
+
+            # checking breakpoints at which to pivot temporary sublist and add it into keys_l
+            if i in breakpoints or i == key_list_options[-1]:
+                keys_l.extend(temp_list[::-1])
+                temp_list = []
+
+        # returning a shuffled keys_l to prevent brute force guessing of shift key
+        return keys_l
+
+    def __make_shift_key(self) -> int:
+        """
+        sum() of the mutated list of ascii values of all characters where the
+        mutated list is the one returned by __neg_pos()
+        """
+        num = sum(self.__neg_pos([ord(x) for x in self.__passcode]))
+        return num if num > 0 else len(self.__passcode)
+
+    def decrypt(self, encoded_message: str) -> str:
+        """
+        Performs shifting of the encoded_message w.r.t. the shuffled __key_list
+        to create the decoded_message
+
+        >>> ssc = ShuffledShiftCipher('4PYIXyqeQZr44')
+        >>> ssc.decrypt("d>**-1z6&'5z'5z:z+-='$'>=zp:>5:#z<'.&>#")
+        'Hello, this is a modified Caesar cipher'
+
+        """
+        decoded_message = ""
+
+        # decoding shift like Caesar cipher algorithm implementing negative shift or reverse shift or left shift
+        for i in encoded_message:
+            position = self.__key_list.index(i)
+            decoded_message += self.__key_list[
+                (position - self.__shift_key) % -len(self.__key_list)
+            ]
+
+        return decoded_message
+
+    def encrypt(self, plaintext: str) -> str:
+        """
+        Performs shifting of the plaintext w.r.t. the shuffled __key_list
+        to create the encoded_message
+
+        >>> ssc = ShuffledShiftCipher('4PYIXyqeQZr44')
+        >>> ssc.encrypt('Hello, this is a modified Caesar cipher')
+        "d>**-1z6&'5z'5z:z+-='$'>=zp:>5:#z<'.&>#"
+
+        """
+        encoded_message = ""
+
+        # encoding shift like Caesar cipher algorithm implementing positive shift or forward shift or right shift
+        for i in plaintext:
+            position = self.__key_list.index(i)
+            encoded_message += self.__key_list[
+                (position + self.__shift_key) % len(self.__key_list)
+            ]
+
+        return encoded_message
+
+
+if __name__ == "__main__":
+    # cip1 = ShuffledShiftCipher()
+    # ciphertext = cip1.encrypt("Hello, this is a modified Caesar cipher")
+    # print(ciphertext)
+    # print(cip1)
+    # print(cip1.decrypt(ciphertext))
```

Comment:
AWESOME WORK!  Just one more ask...
```
def test_end_to_end(msg: str = "Hello, this is a modified Caesar cipher"):
    """
    >>> test_end_to_end()
    """
    Uncomment this code and put it here.
```
This is a function that will run itself in doctests so that we have an end-to-end (roundtrip) test.

Reply:
So will the whole thing look like this:

```
if __name__ == "__main__":
    def test_end_to_end(msg: str = "Hello, this is a modified Caesar cipher"):
        """
        >>> test_end_to_end()
        """
        cip1 = ShuffledShiftCipher()
        ciphertext = cip1.encrypt("Hello, this is a modified Caesar cipher")
        print(ciphertext)
        print(cip1)
        print(cip1.decrypt(ciphertext))
    import doctest

    doctest.testmod()
```

Reply:
Nope.  Move the function OUTSIDE (above) the name/main thing.

Reply:
```
def test_end_to_end(msg: str = "Hello, this is a modified Caesar cipher"):
    """
    >>> test_end_to_end()
    """
    cip1 = ShuffledShiftCipher()
    ciphertext = cip1.encrypt("Hello, this is a modified Caesar cipher")
    print(ciphertext)
    print(cip1)
    print(cip1.decrypt(ciphertext))

if __name__ == "__main__":
    import doctest

    doctest.testmod()
```

Above shows a failed doctest:
```
Failed example:
    test_end_to_end()
Expected nothing
Got:
    a{<</8I+[\*I\*I]I-/`\?\{`I9]{*],I_\:[{,
    Passcode is: kUQ0e8irzElLqvtA1UK
    Hello, this is a modified Caesar cipher
```

Reply:
Am I missing something? @cclauss

Reply:
Made some changes that may have been what was needed. I'm unsure, to be honest. Suggestions welcome.